components/webui/imports/api/search/server/publications.js

```diff
@@ -1,20 +1,21 @@
 import {Meteor} from "meteor/meteor";
 
-import {SEARCH_SERVER_STATE_COLLECTION_NAME} from "../publications";
+import {SEARCH_SERVER_STATE_COLLECTION_NAME, SearchResultsMetadataCollection} from "../publications";
 import {currentServerStateList} from "./query_handler_mediator";
 
 const MyCollections = {};
-const createCollectionsIfNotExist = (sessionId) => {
-    if (!(sessionId in MyCollections)) {
-        MyCollections[sessionId] = {
-            [Meteor.settings.public.SearchResultsCollectionName]: null,
-            [Meteor.settings.public.SearchResultsMetadataCollectionName]: null,
-        };
-        MyCollections[sessionId][Meteor.settings.public.SearchResultsCollectionName] =
-            new Mongo.Collection(`${Meteor.settings.public.SearchResultsCollectionName}_${sessionId}`);
-        MyCollections[sessionId][Meteor.settings.public.SearchResultsMetadataCollectionName] =
-            new Mongo.Collection(`${Meteor.settings.public.SearchResultsMetadataCollectionName}_${sessionId}`);
+const createCollectionsIfNotExist = (sessionId, jobID) => {
+    const collectionName = `${Meteor.settings.public.SearchResultsCollectionName}_${jobID}`;
+
+    if ((MyCollections[sessionId] !== undefined) &&
+        (MyCollections[sessionId][Meteor.settings.public.SearchResultsCollectionName]._name === collectionName)) {
```

Comment:
```suggestion
    if (undefined !== MyCollections[sessionId] &&
        collectionName === MyCollections[sessionId][Meteor.settings.public.SearchResultsCollectionName]._name) {
```
If possible, we should put rvalue on the rhs for comparison